### ISSUE TITLE
Shorthand arrow return appears to be except from linting rules

### ISSUE THREAD

**Issue by justinanastos** [gajus/eslint-plugin-flowtype#15]
With this linting configuration:

```json
    "rules": {
        "flowtype/require-parameter-type": 2,
        "flowtype/require-return-type": [
            2,
            "always",
            {
                "annotateUndefined": "never"
            }
        ],
        "flowtype/space-after-type-colon": [
            2,
            "always"
        ],
        "flowtype/space-before-type-colon": [
            2,
            "never"
        ],
    },
```

The following will not return a linting error:

```javascript
const getUserAgent = (state: Object) => state.device.userAgent;
```

The following will return a false positive:

```javascript
const func = (): Function => (
    () => {}
);
```

```
flowtype/require-return-type   Must not annotate undefined return type
```

The following also returns a false positive:

```javascript

type ISize = {
    width: number,
    height: number,
};
const getScreenDimensions = (state: Object): ISize => state.device.screenDimensions;
```

```
flowtype/require-return-type   Must not annotate undefined return type
```

**Comment by gajus** [comment]
A PR would be welcome. I will not have time to address this in foreseeable future.

**Comment by justinanastos** [comment]
Working on it now 👍

**Comment by justinanastos** [comment]
My apologies for the flood of commit references, I should have added the reference to the PR only.
